packages/plugin-hardhat/src/utils/etherscan-api.ts

```diff
@@ -42,18 +44,17 @@ export async function callEtherscanApi(etherscanApi: EtherscanAPIConfig, params:
  * Makes use of Hardhat Etherscan for handling cases when Etherscan API parameters are not present in config.
  */
 export async function getEtherscanAPIConfig(hre: HardhatRuntimeEnvironment): Promise<EtherscanAPIConfig> {
-  const endpoints = await hre.run('verify:get-etherscan-endpoint');
-  const etherscanConfig: EtherscanConfig = (hre.config as any).etherscan;
-  const key = resolveEtherscanApiKey(etherscanConfig.apiKey, endpoints.network);
-  return { key, endpoints };
+  const endpoints = await getCurrentChainConfig(hre.network, (hre.config as any).etherscan.customChains);
+  const key = resolveEtherscanApiKey((hre.config as any).etherscan.apiKey, endpoints.network);
```

Comment:
Is the `etherscan` field guaranteed to be there?

```suggestion
  const endpoints = await getCurrentChainConfig(hre.network, (hre.config as any).etherscan?.customChains);
  const key = resolveEtherscanApiKey((hre.config as any).etherscan?.apiKey, endpoints.network);
```

Reply:
It should be guaranteed because it is defined in `hardhat-verify`. But since `hardhat-verify` is an optional peer dependency, I think it is safer to explicitly check this for undefined.